src/main/java/io/temporal/samples/hello/HelloPeriodic.java

```diff
@@ -88,14 +96,33 @@
   // Define the workflow implementation which implements the greetPeriodically workflow method.
   public static class GreetingWorkflowImpl implements GreetingWorkflow {
 
+    // In this example, we use an internal of 10 seconds with an intended variation of +/- 2 seconds
+    // between executions of some useful work. In real applications a higher value may be more
+    // appropriate, for example one that matches a business cycle of several hours or even days.
+    private static class ScheduleConfig {
+      static final int PeriodTargetSecs = 10;
+      static final int PeriodVariationSecs = 4;
+    }
+
+    // The max history length of a single Temporal workflow is limited.
+    // Therefore, we cannot loop indefinitely. Instead, we use the ContinueAsNew
```

Comment:
"Therefore, we cannot loop indefinitely" - if possible please be more precise. You can loop indefinitely if in your loop you don't have any blocking operations (like workflow.sleep, activity / child workflow exec, etc). Again, super nice that we add more comments to these samples, but need to make sure that unless we are clear, then its better to not add here and let users ask us on forum/slack.

Reply:
Addressed. Please take a look.